modulemd/tests/test-modulemd-packager-v3.c

```diff
@@ -0,0 +1,234 @@
+  g_assert_nonnull (list);
+  g_assert_null (list[0]);
+  g_clear_pointer (&list, g_strfreev);
+
+
+  list = modulemd_packager_v3_get_rpm_component_names_as_strv (packager);
+  g_assert_nonnull (list);
+  g_assert_null (list[0]);
+  g_clear_pointer (&list, g_strfreev);
+
+  list = modulemd_packager_v3_get_module_component_names_as_strv (packager);
+  g_assert_nonnull (list);
+  g_assert_null (list[0]);
+  g_clear_pointer (&list, g_strfreev);
+}
+
+
+static void
+packager_test_parse_spec (void)
+{
+  MMD_INIT_YAML_EVENT (event);
+  MMD_INIT_YAML_PARSER (parser);
+  g_autofree gchar *yaml_path = NULL;
+  g_autoptr (ModulemdPackagerV3) packager = NULL;
+  g_autoptr (FILE) yaml_stream = NULL;
+  g_autoptr (GError) error = NULL;
+  g_autoptr (ModulemdSubdocumentInfo) subdoc = NULL;
+  g_auto (GStrv) strv = NULL;
+
+  /* Verify a valid YAML file */
+  yaml_path = g_strdup_printf ("%s/yaml_specs/modulemd_packager_v3.yaml",
+                               g_getenv ("MESON_SOURCE_ROOT"));
+  g_assert_nonnull (yaml_path);
+
+  yaml_stream = g_fopen (yaml_path, "rbe");
+  g_assert_nonnull (yaml_stream);
+
+  /* First parse to the subdocument */
+  yaml_parser_set_input_file (&parser, yaml_stream);
+  g_assert_true (yaml_parser_parse (&parser, &event));
+  g_assert_cmpint (YAML_STREAM_START_EVENT, ==, event.type);
+  yaml_event_delete (&event);
+
+  g_assert_true (yaml_parser_parse (&parser, &event));
+  g_assert_cmpint (YAML_DOCUMENT_START_EVENT, ==, event.type);
+  yaml_event_delete (&event);
+
+  subdoc = modulemd_yaml_parse_document_type (&parser);
+  g_assert_nonnull (subdoc);
+  g_assert_no_error (modulemd_subdocument_info_get_gerror (subdoc));
+
+  g_assert_cmpint (MODULEMD_YAML_DOC_PACKAGER,
+                   ==,
+                   modulemd_subdocument_info_get_doctype (subdoc));
+  g_assert_cmpint (MD_PACKAGER_VERSION_THREE,
+                   ==,
+                   modulemd_subdocument_info_get_mdversion (subdoc));
+  g_assert_nonnull (modulemd_subdocument_info_get_yaml (subdoc));
+
+  packager = modulemd_packager_v3_parse_yaml (subdoc, &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (packager);
+  g_assert_true (MODULEMD_IS_PACKAGER_V3 (packager));
+
+
+  g_assert_cmpstr ("foo", ==, modulemd_packager_v3_get_module_name (packager));
+
+  g_assert_cmpstr (
+    "latest", ==, modulemd_packager_v3_get_stream_name (packager));
+
+  g_assert_cmpstr (
+    "An example module", ==, modulemd_packager_v3_get_summary (packager));
+
+  g_assert_cmpstr (
+    "A module for the demonstration of the metadata format. Also, the "
+    "obligatory lorem ipsum dolor sit amet goes right here.",
+    ==,
+    modulemd_packager_v3_get_description (packager));
+
+  strv = modulemd_packager_v3_get_module_licenses_as_strv (packager);
+  g_assert_nonnull (strv);
+  g_assert_nonnull (strv[0]);
+  g_assert_cmpstr ("MIT", ==, strv[0]);
+  g_assert_null (strv[1]);
+  g_clear_pointer (&strv, g_strfreev);
+
```

Comment:
Verify `xmd`?

Reply:
I meant to leave a comment there indicating that verifying XMD would be much easier in a Python test.